src/playlist-loader.js

```diff
@@ -510,6 +508,17 @@ const PlaylistLoader = function(srcUrl, hls, withCredentials) {
           playlist = loader.master.playlists[i];
           loader.master.playlists[playlist.uri] = playlist;
           playlist.resolvedUri = resolveUrl(loader.master.uri, playlist.uri);
+
+          if (!playlist.attributes) {
+            // Although the spec states an #EXT-X-STREAM-INF tag MUST have a
+            // BANDWIDTH attribute, we can play the stream without it. This means a poorly
+            // formatted master playlist may not have an attribute list. An attributes
+            // property is added here to prevent undefined references when we encounter
+            // this scenario.
+            playlist.attributes = {};
+
+            log.warn('Invalid playlist STREAM-INF detected. Missing attribute list.');
```

Comment:
Since `BANDWIDTH` is the only required attribute, maybe it makes sense to just write something like "Invalid playlist STREAM-INF detected. Missing BANDWIDTH attribute."